allspice/utils/list_components.py

```diff
@@ -52,6 +53,8 @@ def list_components(
     repository: Repository,
     source_file: str,
     variant: Optional[str] = None,
+    design_reuse_repos: Optional[list[Repository]] = None,
+    use_first_found_design_reuse_match: Optional[bool] = False,
```

Comment:
I think we can remove this entirely and have it be the default behaviour. If there are multiple, at *most* we should log it and let them know, but having this be an option that throws an error when there are multiple seems confusing to me.

Reply:
If we'd rather keep it, I think we should invert the name and behaviour, so something like:

```suggestion
    require_exactly_one_match: bool = False,
```

We don't need the `Optional` because there's a default value.